### PR TITLE
Get Error From Hessians

### PR DESCRIPTION
**Pull request by varunagrawal**
This PR updates the final elimination DiscreteFactor creation by directly querying the HessianFactors for the error and converting it into a probability.

I also added some minor improvements.

### CODE REVIEW

File: gtsam/hybrid/HybridFactorGraph.cpp [hunk 1]
```diff
@@ -175,27 +175,30 @@ pair<GaussianMixture::shared_ptr, boost::shared_ptr<Factor>> EliminateHybrid(
   DecisionTree<Key, Pair> eliminationResults(sum, eliminate);
 
   // STEP 3: Create result
-  // TODO(Frank): auto pair = eliminationResults.unzip();
+  auto pair = unzip(eliminationResults);
+  GaussianMixture::Conditionals conditionals = pair.first;
```
Comment (dellaert) [inline gtsam/hybrid/HybridFactorGraph.cpp:179]: You are making copies here! Make them const&

File: gtsam/hybrid/HybridFactorGraph.cpp [hunk 2]
```diff
@@ -175,27 +175,30 @@ pair<GaussianMixture::shared_ptr, boost::shared_ptr<Factor>> EliminateHybrid(
...
-  GaussianMixture::Conditionals conditionals(eliminationResults, first);
+  // Create the GaussianMixture from the conditionals
   auto conditional =
       boost::make_shared<GaussianMixture>(keys, discreteKeys, conditionals);
 
   // If there are no more continuous parents, then we should create here a
   // DiscreteFactor, with the error for each discrete choice.
   if (separatorKeys.size() == 0) {
-    auto discreteFactor = factors.toDecisionTreeFactor();
+    auto factorError = [&](const GaussianFactor::shared_ptr& factor) {
```
Comment (dellaert) [inline gtsam/hybrid/HybridFactorGraph.cpp:191]: Create empty VectorValues outside the lambda

**Review by dellaert — APPROVED** [review]
Yah!!!! On to approximate now!

File: gtsam/hybrid/HybridFactorGraph.cpp (hunk 1 above)
Comment (varunagrawal) [reply gtsam/hybrid/HybridFactorGraph.cpp:179]: Ah good catch

Comment (ProfFan) [reply gtsam/hybrid/HybridFactorGraph.cpp]: Forgot to say this in the unzip PR: in C++17 this is really just

`auto [a, b] = unzip(ab)`!

**Review by dellaert — APPROVED** [review]

**Comment by varunagrawal** [comment]
Sorry about that. Just wanted to ensure it was as per your standards. :)

**Comment by varunagrawal** [comment]
CI is failing because someone at Microsoft messed up: https://github.com/dotnet/core/issues/5574

**Comment by dellaert** [comment]
> CI is failing because someone at Microsoft messed up: [dotnet/core#5574](https://github.com/dotnet/core/issues/5574)

Internal PR just merge, work on pruning :-)

**Pull request merged**